Async/Async.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+
+
+/// <summary>
+/// Async is a utility class which provides straight-forward, powerful functions for working with asynchronous C#.
+/// 
+/// Async provides around 20 functions that include the usual 'functional' suspects (map, reduce, filter, each…) as well as
+/// some common patterns for asynchronous control flow (parallel, series, waterfall…). All these functions assume you follow
+/// the convention of providing a single callback as the last argument of your async function.
+/// </summary>
+public class Async {
+	public static void each<T> (List<T> items, Action<T, Action<Exception>> fn, Action<Exception> cb) {
+		if (items == null || items.Count == 0) {
+			cb(null);
+			return;
+		}
+
+		int currentItemI = 0;
+		Action iterate = null;
+		iterate = () => {
+			if (currentItemI >= items.Count) {
+				cb(null);
+				return;
+			}
+
+			// Execute the given function on this item
+			fn(items[currentItemI], (Exception error) => {
+				// Stop iteration if there was an error
+				if (error != null) {
+					cb(error);
+					return;
+				}
+
+				// Continue to next item
+				currentItemI++;
+				iterate();
```

Comment:
By Coroutines you mean making this library class depend on Unity?  I'm not sure thats a good idea, if it's even possible? (main thread problems?)